cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -219,6 +248,16 @@ private void executeAction(ApplicationSpecification appSpec, WorkflowActionNode
     status.put(node.getNodeId(), node);
 
     final WorkflowAction action = initialize(actionSpec, classLoader, instantiator, token, node.getNodeId());
+
+    final TransactionContext transactionContext = new TransactionContext(txClient);
```

Comment:
Should I have every method each get all the datasets again and create a new TransactionContext?